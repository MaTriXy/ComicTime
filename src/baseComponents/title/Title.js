import React from "react";
import DocumentTitle from "react-document-title";
import { translateComponent } from "../../i18n/i18n";
import { connect } from "react-redux";
import { Maybe } from "monet";

class Title extends React.Component {
  render() {
    const { t } = this.props;
    const title = Maybe.fromNull(this.props.title).orSome(t("appName"));
    return <DocumentTitle title={title}>{this.props.children}</DocumentTitle>;
  }
}

Title.defaultProps = {
  title: undefined
};

export function mapStateToProps(state) {
  const customTitle = state.publicationDetail.publicationSelected
    .map(publication => publication.title)
    .orSome(undefined);
  return {
    title: customTitle
  };
}

export const PureTitle = Title;
export default translateComponent(connect(mapStateToProps, null)(PureTitle));
