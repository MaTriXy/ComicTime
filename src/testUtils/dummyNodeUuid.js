jest.mock("node-uuid", () => {
  return {
    v4() {
      return "uuid";
    }
  };
});
