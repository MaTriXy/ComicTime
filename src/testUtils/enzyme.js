import React from "react";
import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { MuiThemeProvider } from "material-ui";
import comicTimeTheme from "../theme";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

export function mockHistory() {
  return {
    go: jest.fn(),
    goBack: jest.fn()
  };
}

export function translator(key) {
  return key;
}

export function mountWithMuyThemeProvider(component) {
  return Enzyme.mount(
    <MuiThemeProvider muiTheme={comicTimeTheme}>{component}</MuiThemeProvider>
  );
}

export function mountComponent(component) {
  return Enzyme.mount(component);
}

export function mountComponentAsJson(component) {
  return toJson(mountComponent(component));
}

export function shallowComponent(component) {
  return Enzyme.shallow(component);
}

export function shallowComponentAsJson(component) {
  return toJson(shallowComponent(component));
}

export function json(component) {
  return toJson(component);
}
