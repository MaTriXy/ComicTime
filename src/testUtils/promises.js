export async function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}
