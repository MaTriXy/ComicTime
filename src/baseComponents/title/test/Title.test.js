jest.mock("../../../i18n/i18n");

import { PureTitle } from "../Title";
import React from "react";
import { shallowComponentAsJson, translator } from "../../../testUtils/enzyme";

describe("Title", () => {
  it("shows the app title by default if it's not configured", () => {
    const component = shallowComponentAsJson(<PureTitle t={translator} />);
    expect(component).toMatchSnapshot();
  });

  it("shows the title prop instead of the app name if it's configured properly", () => {
    const component = shallowComponentAsJson(
      <PureTitle t={translator} title={"The Flash #1"} />
    );
    expect(component).toMatchSnapshot();
  });
});
