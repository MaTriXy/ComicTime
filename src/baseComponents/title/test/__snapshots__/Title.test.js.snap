// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title shows the app title by default if it's not configured 1`] = `
<SideEffect(DocumentTitle)
  title="appName"
/>
`;

exports[`Title shows the title prop instead of the app name if it's configured properly 1`] = `
<SideEffect(DocumentTitle)
  title="The Flash #1"
/>
`;
